apps/nextjs-website/src/app/auth/confirmation/page.tsx

```diff
@@ -0,0 +1,57 @@
+'use client';
+import { Box, CircularProgress, Stack } from '@mui/material';
+import { environment } from '@/config';
+import PageNotFound from '@/app/not-found';
+import { Auth } from 'aws-amplify';
+import { useSearchParams, useRouter } from 'next/navigation';
+import { useEffect, useState } from 'react';
+
+const Confirmation = () => {
+  const searchParams = useSearchParams();
+  const router = useRouter();
+  const username = searchParams.get('username');
+  const code = searchParams.get('code');
+
+  const [renderNotFound, setRenderNotFound] = useState(environment === 'prod');
+
+  useEffect(() => {
+    if (username && code) {
+      Auth.confirmSignUp(username, code)
+        .then(() => {
+          setRenderNotFound(false);
+          // eslint-disable-next-line functional/immutable-data
+          router.push('/auth/account-activated');
+        })
+        .catch(() => {
+          setRenderNotFound(true);
+        });
+    } else {
+      setRenderNotFound(true);
+    }
+  }, [username, code]);
+
+  return renderNotFound ? (
+    <PageNotFound />
```

Comment:
This should be replaced with a more specific error page. I think we should add a TODO